### PR TITLE
Implement MultiTargetAnnotation

### PR DESCRIPTION
**Pull request by sequencer**
I'm writing a physical design annotation library, which needs `MultiTargetAnnotation` for multi signal annotation. Since there are no historic issues, `MultiTargetAnnotation` defines targets as `Seq[Target]`.

### CODE REVIEW

File: src/main/scala/firrtl/annotations/Annotation.scala [hunk 1]
```diff
@@ -77,6 +77,28 @@ trait SingleTargetAnnotation[T <: Named] extends Annotation {
   }
 }
 
+trait MultiTargetAnnotation extends Annotation {
```
Comment (azidar) [inline src/main/scala/firrtl/annotations/Annotation.scala]: This needs ScalaDoc describing its behavior.

In addition, what should the desired behavior be if the two targets are renamed to a different number of new targets? I wonder if there is a better way to represent this, perhaps as a Seq[Seq[Target]], where each target is expanded (flatMap) within its container Seq?

E.g. Seq(Seq(a), Seq(b)) -> Seq(Seq(a1, a2), Seq(b1, b2, b3))

Food for thought. I agree though that we should certainly add something like this - I've found myself duplicating this code over and over again....

Comment (sequencer) [reply src/main/scala/firrtl/annotations/Annotation.scala]: I try to folded `Seq[Seq[Target]]` to `Seq[Target]` is aimed to give a same behavior with `SingleTargetAnnotation`. If you permit `Seq[Seq[Target]]` would be much better.
 I think `crossJoin` is also a useful function for the multi target folding in `RenameMap`, so do mind I keeping them as a private function with detail documentation?

**Comment by azidar** [comment]
This LGTM now, @jackkoenig any thoughts before I merge?

**Comment by azidar** [comment]
Also, yeah I think crossJoin is a good idea, feel free to add it as a private function with documentation.

**Comment by sequencer** [comment]
done.

File: src/main/scala/firrtl/annotations/Annotation.scala [hunk 2]
```diff
@@ -77,6 +77,48 @@ trait SingleTargetAnnotation[T <: Named] extends Annotation {
   }
 }
 
+/** Annotate Multi Target in a single Annotation,
+  * rather than [[SingleTargetAnnotation]] which duplicate the [[Annotation]] based on the [[RenameMap]],
+  * [[MultiTargetAnnotation]] keeps a [[RenameMap]] in the targets for better data structure
```
Comment (azidar) [inline src/main/scala/firrtl/annotations/Annotation.scala]: Reword to: "[[MultiTargetAnnotation]] keeps the renamed targets grouped within a single annotation."

Comment (sequencer) [reply src/main/scala/firrtl/annotations/Annotation.scala]: Sorry for my redundant documentations.

**Pull request merged**